File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        arr=[]
        count=0
        for i in range(0, n + 1):
            for j in range(0,32):
                if(i&1==1):
                    count+=1
                i >>= 1
            arr.append(count)
            count=0
        return arr

File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> ans(n + 1);
        for (int i = 0; i <= n; i++)
        {
            int temp = i;
            for (int j = 0; j < 32; j++)
            {
                if(temp & 1)
                    ans[i]++;
                temp >>= 1;
            }
        }
        return ans;
    }
};